basic_autonomy/include/basic_autonomy/helper_functions.h

```diff
@@ -95,7 +95,18 @@ namespace waypoint_generation
      * \return index of nearest point in points
      */
     int get_nearest_index_by_downtrack(const std::vector<lanelet::BasicPoint2d>& points, const carma_wm::WorldModelConstPtr& wm,
-                                      const cav_msgs::VehicleState& state);                                                                                                                               
+                                      const cav_msgs::VehicleState& state);    
+
+    /**
+     * \brief Downsamples an input lanelet::BasicPoint2d vector by saving only each nth element.
+     * For example, given an input vector of { 0, 1, 2, 3, 4, 5 } and n = 2
+     * The output vector will be {0, 2, 4}
+     * 
+     * \param input The input lanelet::BasicPoint2d vector to downsample
+     * \param n The count of the elements to save
+     * \return The downsampled vector
+     */ 
+    std::vector<lanelet::BasicPoint2d> downsample_basicpoint2d_vector(const std::vector<lanelet::BasicPoint2d>& input, unsigned int n);                                                                                                                           
```

Comment:
What does this method do that the downsample method from carma-utils does not? The utils method is templated and therefore can support any type of vector, this seems like an unnecessary specialization?